src/script/script.cpp

```diff
@@ -258,9 +257,10 @@ vector<unsigned char> CScript::AddressHash() const
     } else if (this->IsPayToPublicKeyHash()) {
         start = 3;
     } else {
-        assert(false);
+        // unknown script type; return empty vector
+        vector<unsigned char> hashBytes;
+        return hashBytes;
```

Comment:
The spentindex changes need to call this `AddressHash()` method even if we don't recognize this script type, and returning an empty vector is what's needed (matches upstream behavior).